isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp

```diff
@@ -0,0 +1,245 @@
+    return 1;
+  }
+
+  QList<QString> ControlNetVitals::getCubeSerials() {
+    return m_controlNet->GetCubeSerials();
+  }
+
+  QList<ControlPoint*> ControlNetVitals::getAllPoints() {
+    return m_controlNet->GetPoints();
+  }
+
+  // REFACTOR
+  QList<ControlPoint*> ControlNetVitals::getIgnoredPoints() {
+    QList<ControlPoint*> ignoredPoints;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->IsIgnored()) ignoredPoints.append(point);
+    }
+    return ignoredPoints;
+  }
+
+  QList<ControlPoint*> ControlNetVitals::getLockedPoints() {
+    QList<ControlPoint*> lockedPoints;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->IsEditLocked()) lockedPoints.append(point);
+    }
+    return lockedPoints;
+  }
+
+  QList<ControlPoint*> ControlNetVitals::getFixedPoints() {
+    QList<ControlPoint*> fixedPoints;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->GetPointTypeString() == "Fixed") fixedPoints.append(point);
+    }
+    return fixedPoints;
+  }
+
+  QList<ControlPoint*> ControlNetVitals::getConstrainedPoints() {
+    QList<ControlPoint*> constrainedPoints;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->GetPointTypeString() == "Constrained") constrainedPoints.append(point);
+    }
+    return constrainedPoints;
+  }
+
+  QList<ControlPoint*> ControlNetVitals::getFreePoints() {
+    QList<ControlPoint*> freePoints;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->GetPointTypeString() == "Free") freePoints.append(point);
+    }
+    return freePoints;
+  }
+
+  // REFACTOR
+  QList<ControlPoint*> ControlNetVitals::getPointsBelowMeasureThreshold(int num) {
+    QList<ControlPoint*> belowThreshold;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->GetNumMeasures() < num) belowThreshold.append(point);
+    }
+    return belowThreshold;
+  }
+
+  QList<QString> ControlNetVitals::getAllImageSerials() {
+    return m_controlNet->GetCubeSerials();
+  }
+
+  // REFACTOR
+  QList<QString> ControlNetVitals::getImagesBelowMeasureThreshold(int num) {
+    QList<QString> imagesBelowThreshold;
+    foreach(QString serial, m_controlNet->GetCubeSerials()) {
+      if (m_controlNet->GetMeasuresInCube(serial).size() < num) imagesBelowThreshold.append(serial);
+    }
+    return imagesBelowThreshold;
+  }
+
+  // REFACTOR
+  QList<QString> ControlNetVitals::getImagesBelowHullTolerance(int num) {
+    QList<QString> list;
+    list.append("Example.cub");
+    return list;
+  }
+
+  QString ControlNetVitals::getStatus() {
+    return m_status;
+  }
+
+  QString ControlNetVitals::getStatusDetails() {
+    return m_statusDetails;
+  }
+
+  QString ControlNetVitals::getNetworkId() {
+    return m_controlNet->GetNetworkId();
+  }
+
+  //
+  // ImageVitals ControlNetVitals::getImageVitals(QString serial) {
+  //   return NULL;
+  // }
+
+  void ControlNetVitals::validate() {
```

Comment:
This method looks like GUI functionality. Is there a reason why it's not in the GUI?

Reply:
I felt like the overall status of a network was relevant information to store in the NetworkVitals object to reflect the current sitting status of it's Control Network. This slot is connected to the netModified() slot of the ControlNet to recalculate the overall status of the network. If we decide that the status and health of a network should be determined by the GUI then I can absolutely move it there

Reply:
Sounds good